### ISSUE TITLE
acceptAllCerts on iOS not working

### ISSUE THREAD

**Issue by SirTori** [silkimen/cordova-plugin-advanced-http#29]
When setting acceptAllCerts to true every request via https returns an error with  `{"status": -1, "error": "cancelled"}`

my code

```javascript
if(isDevMode()) {
  this.http.acceptAllCerts(true).then(
    () => console.warn("ALL CERTS WILL BE ACCEPTED!"),
    () => console.error("ERROR SETTING acceptAllCerts")
  );
}
```
During development I used to set the option acceptAllCerts  to true.
I migrated to ionic 3.7 and from https://github.com/Telerik-Verified-Plugins/SecureHTTP to this plugin (version 1.6.1) as this is now the official ionic-native http plugin and has all the features that made me use SecureHTTP instead of the previous official ionic-native http plugin.
With the old plugin and ionic 3.6 it was never a problem on iOS 10 and after upgrading suddenly nothing worked anymore. Took me a few hours to identify acceptAllCerts as the culprit.

As a workaround I just do no longer use acceptAllCerts in development and test against servers with "proper" certificates.

It would be nice for this to get fixed, or at least have a comment in the docs warning that the option won't work on iOS

**Comment by silkimen** [comment]
Hi SirTori, I'm sorry for the inconvenience. But I'm curious, did you also run `validateDomainName(false)`?

e.g.
```js
if(isDevMode()) {
  this.http.acceptAllCerts(true)
     .then(() => this.http.validateDomainName(false))
     .then(() => ... send your request ...)
     .catch(console.error)
  );
}
```

**Comment by SirTori** [comment]
Hi silkimen,
no I did not - wasn't neccessary before

**Comment by silkimen** [comment]
Could you please try running my example code against your server? I'd like to know if this fixes your problem.

**Comment by silkimen** [comment]
Could you please try running my example code against your server? I'd like to know if this fixes your problem.

**Comment by SirTori** [comment]
Thanks for looking into it.

I just tried it and together with `this.http.validateDomainName(false)` it seems to work just fine.

**Comment by silkimen** [comment]
OK, thank you!

I think we'll change the interface and remove `validateDomainName` because you are not the first one having problems with this one. We'll disable domain name validation automatically when running `acceptAllCerts(true)` instead.

**Comment by silkimen** [comment]
@SirTori your problem should be fixed in v1.6.2

**Comment by tuakapro** [comment]
I have same issue in ios, always return code -999 and request cancelled when i use http.post or any request, only fail in ios but working fine in android
